### ISSUE TITLE
[ENHANCEMENT] information of CMakeSettings.json

### ISSUE THREAD

**Issue by orion160** [lukka/run-cmake#39]
When run-cmake starts there is information about the build, that information does not correspond to the CMakeSettings.json and is to be overrided in the json file.

![image](https://user-images.githubusercontent.com/61389980/103382986-12d3a000-4abf-11eb-970c-671e60d596eb.png)

And the CMakeSettings.json  doesnt show information about the config

![image](https://user-images.githubusercontent.com/61389980/103382939-ea4ba600-4abe-11eb-902f-ed73dc8fe8ec.png)

Well, i think that these features are under developmemnt.

**Comment by lukka** [comment]
@orion160 
1 - The output shown in there is emitted by the github runner, there is nothing it can done to make it right. The values shown in there must be disregarded accordingly to the current mode. For example, if the input `cmakeListsOrSettingsJson` is set to `CMakeSettingsJson`, only the _valid_ input (as specified in [actions.yml](https://github.com/lukka/run-cmake/blob/main/action.yml)) are taken into account by the action, all the others are disregarded.
A solution to this would be to split this action in three different actions. Having a "mode selection" is pretty much a burden I would avoid indeed, and it is not a good idea indeed.

2 - When using CMakeSettings.json, In the output of the `run-cmake` each configuration name is printed out while it is being processed, e.g.:
   `Process configuration: 'x64-Debug'.`
If it is not printed in your workflow it may mean nothing matched the regular expression defined in `configurationRegexFilter` input.

**Comment by orion160** [comment]
Thanks.
What i was referring, is where i can see the configuration that CMakeSettings.json overrides, or send to cmake executable.
Of course the regex filter is working, but where do i know the actual parameters that are sent to CMake, as the first image shows.

For the moment the information that run-cmake shows to me is 

![image](https://user-images.githubusercontent.com/61389980/103465171-53255f00-4d07-11eb-874f-cacc94a8c6a7.png)

As #38 there are some settings that are discarded to run cmake, the cmakesettings.json schema dictates  [this parameters](https://docs.microsoft.com/en-us/cpp/build/cmakesettings-reference?view=msvc-160) for which some of them are passed to Visual Studio and Others to cmake. 

The problem urges where the documentation doesnt specifies which is passed to cmake, and so use it in run-cmake.

**Comment by orion160** [comment]
As an example, [Action.yml](https://github.com/lukka/run-cmake/blob/main/action.yml)



![image](https://user-images.githubusercontent.com/61389980/103465492-eeb7cf00-4d09-11eb-8aa9-eb240728d519.png)

gives this information, which means that when i am using CMakeSettings.json that information doesnt matter.

but as it was shown in the previous comment, the information about the cmakebuild type is not echoed when processing the config, so i cant certainly know if it is gathering the cmakebuildtype from the json file

**Comment by lukka** [comment]
@orion160 yes, the output in case of the CMakeSettings.json could be improved. Right now the only way to see more verbose log (but way too much) is to enable [the verbose long indeed](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging#enabling-step-debug-logging) :)

Note that file will be replaced by the [CMakePresets.json](https://cmake.org/cmake/help/v3.19/manual/cmake-presets.7.html), which will be consumed directly by CMake, hence ton of code will be dropped out of this action soon, hurray!

**Comment by orion160** [comment]
@lukka that means that CMakeSettings.json will be deprecated or eliminated?

**Comment by lukka** [comment]
@orion160 it will, yes, removed from run-cmake. Yet the existing versions of this action are and will be supporting it.
